pkg/await/apps_statefulset_test.go

```diff
@@ -0,0 +1,1252 @@
+				}},
+		},
+		{
+			description: "[Revision 1] Failure should only report Pods from active StatefulSet, part 1",
+			do: func(statefulsets, pods chan watch.Event, timeout chan time.Time) {
+				podName := "foo-0"
+
+				statefulsets <- watchAddedEvent(
+					statefulsetProgressing(inputNamespace, inputName, targetService))
+
+				// Ready Pod should generate no errors.
+				pods <- watchAddedEvent(statefulsetReadyPod(inputNamespace, podName, inputName))
+
+				// Pod belonging to some other StatefulSet should not show up in the errors.
+				pods <- watchAddedEvent(statefulsetFailedPod(inputNamespace, podName, "bar"))
+
+				// Timeout. Failure.
+				timeout <- time.Now()
+			},
+			expectedError: &timeoutError{
+				object: statefulsetProgressing(inputNamespace, inputName, targetService),
+				subErrors: []string{
+					"Failed to observe the expected number of ready replicas",
+				}},
+		},
+		{
+			description: "[Revision 2] Failure should only report Pods from active StatefulSet, part 1",
+			do: func(statefulsets, pods chan watch.Event, timeout chan time.Time) {
+				podName := "foo-0"
+
+				statefulsets <- watchAddedEvent(
+					statefulsetUpdating(inputNamespace, inputName, targetService))
+
+				// Ready Pod should generate no errors.
+				pods <- watchAddedEvent(statefulsetReadyPod(inputNamespace, podName, inputName))
+
+				// Pod belonging to some other StatefulSet should not show up in the errors.
+				pods <- watchAddedEvent(statefulsetFailedPod(inputNamespace, podName, "bar"))
+
+				// Timeout. Failure.
+				timeout <- time.Now()
+			},
+			expectedError: &timeoutError{
+				object: statefulsetUpdating(inputNamespace, inputName, targetService),
+				subErrors: []string{
+					"Failed to observe the expected number of ready replicas",
+					".status.currentRevision does not match .status.updateRevision",
+				}},
+		},
+		{
+			description: "[Revision 1] Failure should only report Pods from active StatefulSet, part 2",
+			do: func(statefulsets, pods chan watch.Event, timeout chan time.Time) {
+				podName := "foo-0"
+
+				statefulsets <- watchAddedEvent(
+					statefulsetProgressing(inputNamespace, inputName, targetService))
+
+				// Failed Pod should show up in the errors.
+				pods <- watchAddedEvent(statefulsetFailedPod(inputNamespace, podName, inputName))
+
+				// Unrelated successful Pod should generate no errors.
+				pods <- watchAddedEvent(statefulsetReadyPod(inputNamespace, podName, "bar"))
+
+				// Timeout. Failure.
+				timeout <- time.Now()
+			},
+			expectedError: &timeoutError{
+				object: statefulsetProgressing(inputNamespace, inputName, targetService),
+				subErrors: []string{
+					"Failed to observe the expected number of ready replicas",
+					"1 Pods failed to run because: [ErrImagePull] manifest for nginx:busted not found",
+				}},
+		},
+		{
+			description: "[Revision 2] Failure should only report Pods from active StatefulSet, part 2",
```

Comment:
This one, on the other hand, has 2 part 2's?

Reply:
Both cases are covered, but the ordering may not have been what you expected. First two don't have related Pod errors, second two tests do.